### ISSUE TITLE
Inline evaluation display renders differently than the REPL display for empty spaces

### ISSUE THREAD

**Issue by nikijiandani** [BetterThanTomorrow/calva#872]
`
(ns dojo.exercise-1
  (:require [clojure.string :as string]))

(defn pad [string n]
  (apply str string (repeat (- n (count string)) " ")))

(pad "Hello" 10)
`

When I evaluate the code above, I see the correct result in the output repl
`"Hello     "`
However the inline view in the code editor shows the result with an incorrect number of spaces
`"Hello "`

**Comment by rafd** [comment]
Even more simply: 
![Screenshot from 2020-12-17 21-57-02](https://user-images.githubusercontent.com/89664/102569150-e8431980-40b2-11eb-8321-0fa6002292b1.png)

**Comment by rafd** [comment]
Sadly, looks like `createTextEditorDecorationType` doesn't allow for `white-space` attribute. https://code.visualstudio.com/api/references/vscode-api#DecorationRenderOptions

**Comment by bpringe** [comment]
Thanks for the report. It does appear that whitespace is allowed at the end of text in decorations, though, since we see some, but not all in the above examples. I can't tell but is each result showing only 1 space at the end? Maybe all are getting truncated to 1 space somehow.

A PR on this would be very welcome if anyone has the time.

**Comment by PEZ** [comment]
I think it is just how css works. Those are `:before` `content`.  https://developer.mozilla.org/en-US/docs/Web/CSS/content

We might get away by replacing strings of spaces with some unicode whitespace.

**Comment by rafd** [comment]
To clarify @bpringe, the issue is not related to beginning-of-string or end-of-string spaces, but that browsers collapse multiple consecutive spaces into one (except when using `<pre>` or `whitespace: pre` in CSS, which, the code segments in VSCode _do_ use, but the annotations do not). 

If we could add `whitespace: pre` to the CSS for that element, it would respect whitespace, but it's not something that VSCode allows (perhaps this should be an upstream request to VSCode?). 

As @PEZ suggested, a bandaid solution could be to replace all spaces with a [non-breaking space](https://www.fileformat.info/info/unicode/char/00a0/index.htm). 

 ```clojure 
(clojure.string/replace "    text     " " " "\u00a0")
```

The only catch being: would it then confuse certain users if they copy-pasted and found non-breaking-spaces instead of regular spaces? In my opinion, as long as the replacement is limited to the annotation, and not the REPL output, it should be OK.

**Comment by PEZ** [comment]
Indeed. It's not even possible to select and copy the inline display.

**Comment by bpringe** [comment]
Thanks for the info @rafd. Sounds like a good approach.
